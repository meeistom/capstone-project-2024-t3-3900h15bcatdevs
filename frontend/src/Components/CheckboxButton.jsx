import React from "react";
import { ToggleButton } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSquare, faCheckSquare } from "@fortawesome/free-regular-svg-icons";

import "bootstrap/dist/css/bootstrap.min.css";
import "../index.css";

export { CheckboxButton };

function CheckboxButton({ isChecked, onChange, name, children }) {
  console.log(isChecked, name, onChange);
  return (
    <>
      <ToggleButton
        type="checkbox"
        className={isChecked ? "check" : "uncheck"}
        checked={isChecked}
        name={name}
        value={name}
        onChange={onChange}
      >
        {isChecked ? (
          <FontAwesomeIcon icon={faCheckSquare} />
        ) : (
          <FontAwesomeIcon icon={faSquare} />
        )}
        <p className="check-btn-child" style={{ marginLeft: "10px" }}>
          {children}
        </p>
      </ToggleButton>
    </>
  );
}
