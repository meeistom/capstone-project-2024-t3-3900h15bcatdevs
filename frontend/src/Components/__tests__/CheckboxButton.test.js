import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { CheckboxButton } from "../CheckboxButton";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render checkbox button", () => {
  render(
    <CheckboxButton
      data-testid="test-checkbox"
      id="test-checkbox"
      name="test-checkbox"
      aria-label="test-checkbox"
      isChecked={true}
      onChange={() => {}}
    >
      Am checked
    </CheckboxButton>
  );
  const checkboxButton = screen.getByRole("checkbox", { name: /am checked/i });

  expect(checkboxButton).toBeInTheDocument();
});

test("should match snapshot when checked", () => {
  const tree = renderer
    .create(
      <CheckboxButton
        data-testid="test-checkbox"
        id="test-checkbox"
        name="test-checkbox"
        aria-label="test-checkbox"
        isChecked={true}
        onChange={() => {}}
      >
        Am unchecked
      </CheckboxButton>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should match snapshot when unchecked", () => {
  const tree = renderer
    .create(
      <CheckboxButton
        data-testid="test-checkbox"
        id="test-checkbox"
        name="test-checkbox"
        aria-label="test-checkbox"
        isChecked={false}
        onChange={() => {}}
      >
        Not Checked
      </CheckboxButton>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should pass down children", () => {
  render(
    <CheckboxButton
      data-testid="test-checkbox"
      id="test-checkbox"
      name="test-checkbox"
      aria-label="test-checkbox"
      isChecked={true}
      onChange={() => {}}
    >
      Am checked
    </CheckboxButton>
  );
  const labelText = screen.getByText("Am checked");
  expect(labelText).toBeInTheDocument();
});

test("should handle onClick", async () => {
  const onClick = jest.fn();

  render(
    <CheckboxButton
      data-testid="test-checkbox"
      id="test-checkbox"
      name="test-checkbox"
      aria-label="test-checkbox"
      isChecked={true}
      onChange={onClick}
    >
      Am checked
    </CheckboxButton>
  );
  const checkboxButton = screen.getByRole("checkbox", { name: /am checked/i });
  fireEvent.click(checkboxButton);

  expect(onClick).toHaveBeenCalledTimes(1);
});
