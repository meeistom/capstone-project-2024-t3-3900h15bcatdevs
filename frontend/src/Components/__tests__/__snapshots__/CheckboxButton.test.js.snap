// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when checked 1`] = `
Array [
  <input
    autoComplete="off"
    checked={true}
    className="btn-check"
    disabled={false}
    id="test-checkbox"
    name="test-checkbox"
    onChange={[Function]}
    type="checkbox"
  />,
  <label
    className="check btn btn-primary"
    disabled={false}
    htmlFor="test-checkbox"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-square-check "
      data-icon="square-check"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 80c-8.8 0-16 7.2-16 16l0 320c0 8.8 7.2 16 16 16l320 0c8.8 0 16-7.2 16-16l0-320c0-8.8-7.2-16-16-16L64 80zM0 96C0 60.7 28.7 32 64 32l320 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L64 480c-35.3 0-64-28.7-64-64L0 96zM337 209L209 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L303 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="ms-1"
    >
      Am unchecked
    </span>
  </label>,
]
`;

exports[`should match snapshot when unchecked 1`] = `
Array [
  <input
    autoComplete="off"
    checked={false}
    className="btn-check"
    disabled={false}
    id="test-checkbox"
    name="test-checkbox"
    onChange={[Function]}
    type="checkbox"
  />,
  <label
    className="uncheck btn btn-primary"
    disabled={false}
    htmlFor="test-checkbox"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-square "
      data-icon="square"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 80c8.8 0 16 7.2 16 16l0 320c0 8.8-7.2 16-16 16L64 432c-8.8 0-16-7.2-16-16L48 96c0-8.8 7.2-16 16-16l320 0zM64 32C28.7 32 0 60.7 0 96L0 416c0 35.3 28.7 64 64 64l320 0c35.3 0 64-28.7 64-64l0-320c0-35.3-28.7-64-64-64L64 32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="ms-1"
    >
      Not Checked
    </span>
  </label>,
]
`;
